### ISSUE TITLE
IoT Edge Bad performance in simple route (and measuring RTT)

### ISSUE THREAD

**Issue by azthec** [Azure/iotedge#1643]
Hello I am currently evaluating the IoT Hub and IoT Edge, part of these evaluations are performance readings of message RTT, and IoT Edge seems to fail completely even at a measly 10 Messages per Second. I have tried many variations of code, settings (OptimizeForPerformance),  transport (AMQP, MQTT...), hardware, operating systems, you name it. The messages are being sent and timed concurrently.

When testing directly with the Hub it even outperforms the documentation, the free Hub reads "Higher of 100 send operations/sec or 12 send operations/sec/unit " and I have had no issues, even with a single unit at 160 messages per second.  When testing with the Edge server with 1000 messages sent at 5 Messages per Second the last message arrives with a delay of over 2 minutes since it was initially sent.

Is there some setting or configuration that I am missing? The device load, memory, IO and network usage barely changes during testing, leading me to conclude this bottleneck is somewhere on the software or the testing method.

Possibly related to #44 

## Expected Behavior
For Edge to be viable a large number of messages per second must be supported.

## Current Behavior
Currently the testing barely supports a single device.

## Steps to Reproduce

1. Start up a Edge server with a simple route to upstream rule.
2. Register a consumer for received Hub messages.
3. Send D2C messages, going through the Edge server.
4. Time the messages flight time via timestamps.

### Output of `iotedge check`

<details>
<summary>Click here</summary>

```
// Paste here
```
</details>

### Device (Host) Operating System
Ubuntu 16.04 or Windows 10

### Architecture
amd64

### Container Operating System
Linux containers or Windows containers

#### iotedged
iotedge 1.0.6.1 (3fa6cbef8b7fc3c55a49a622735eb1021b8a5963)

#### Docker

<details><summary>CLICK ME</summary>
<p>

```
Client:
 Version:           3.0.3
 API version:       1.40
 Go version:        go1.11.4
 Git commit:        48bd4c6d
 Built:             Wed Jan 23 16:17:56 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          3.0.4
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.11.4
  Git commit:       8ecd530
  Built:            Fri Jan 25 01:45:38 2019
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.2.2
  GitCommit:        9754871865f7fe2f4e74d43e2fc7ccd237edcbce
 runc:
  Version:          1.0.0-rc6+dev
  GitCommit:        96ec2177ae841256168fcf76954f7177af9446eb
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```
</p>
</details>

## Logs
There are no errors on any of the logs.

**Comment by varunpuranik** [comment]
Hi @azthec - this was a known issue in EdgeHub which has since been fixed, as @darobs has explained. Please try EdgeHub version 1.0.8.1 - the issue should go away.

**Comment by azthec** [comment]
This does indeed seem to be fixed in version 1.0.8.1, as does the Node SDK AQMP client not working through Edge.

Closing.
